tsdb/shard.go

```diff
@@ -472,6 +506,11 @@ func (s *Shard) SeriesKeys(opt influxql.IteratorOptions) (influxql.SeriesList, e
 		return []influxql.Series{{Aux: auxFields}}, nil
 	}
 
+	s.mu.RLock()
```

Comment:
`defer` does do an allocation, and testing locally, is about ~4x slower than a direct `Lock()` then `Unlock()`. If you know exactly where the code is going to return, and aren't worried about cleaning up during a `panic()`, then doing this is better for code that will be called often:

```
func (s *Shard) IsClosed() bool {
  s.mu.RLock()
  closed := s.engine == nil
  s.mu.RUnlock()
  return closed
}
```